app/src/main/java/agency/five/codebase/android/movieapp/ui/home/HomeScreen.kt

```diff
@@ -0,0 +1,315 @@
+    MovieCategory.UPCOMING_THIS_WEEK
+)
+
+private val popularCategoryViewState = homeScreenMapper.toHomeMovieCategoryViewState(
+    popularCategoryLabels,
+    MovieCategory.POPULAR_STREAMING,
+    MoviesMock.getMoviesList()
+)
+private val nowPlayingCategoryViewState = homeScreenMapper.toHomeMovieCategoryViewState(
+    nowPlayingCategoryLabels,
+    MovieCategory.NOW_PLAYING_MOVIES,
+    MoviesMock.getMoviesList()
+)
+private val upcomingCategoryViewState = homeScreenMapper.toHomeMovieCategoryViewState(
+    upcomingCategoryLabels,
+    MovieCategory.UPCOMING_TODAY,
+    MoviesMock.getMoviesList()
+)
+
+@Composable
+fun HomeScreenRoute(
+    onNavigateToMovieDetails: (Int) -> Unit
+) {
+    var popularViewState by remember { mutableStateOf(popularCategoryViewState) }
+    var nowPlayingViewState by remember { mutableStateOf(nowPlayingCategoryViewState) }
+    var upcomingViewState by remember { mutableStateOf(upcomingCategoryViewState) }
+
+    HomeScreen(
+        popularViewState = popularViewState,
+        nowPlayingViewState = nowPlayingViewState,
+        upcomingViewState = upcomingViewState,
+        onPopularLabelClick = {
+            popularViewState = homeScreenMapper.toHomeMovieCategoryViewState(
+                popularCategoryLabels,
+                MovieCategory.getByOrdinal(it),
+                MoviesMock.getMoviesList()
+            )
+        },
+        onNowPlayingLabelClick = {
+            nowPlayingViewState = homeScreenMapper.toHomeMovieCategoryViewState(
+                nowPlayingCategoryLabels,
+                MovieCategory.getByOrdinal(it),
+                MoviesMock.getMoviesList()
+            )
+        },
+        onUpcomingLabelClick = {
+            upcomingViewState = homeScreenMapper.toHomeMovieCategoryViewState(
+                upcomingCategoryLabels,
+                MovieCategory.getByOrdinal(it),
+                MoviesMock.getMoviesList()
+            )
+        },
+        onNavigateToMovieDetails = onNavigateToMovieDetails,
+        modifier = Modifier
+    )
+}
+
+@Composable
+fun HomeScreen(
+    popularViewState: HomeMovieCategoryViewState,
+    nowPlayingViewState: HomeMovieCategoryViewState,
+    upcomingViewState: HomeMovieCategoryViewState,
+    onPopularLabelClick: (Int) -> Unit,
+    onNowPlayingLabelClick: (Int) -> Unit,
+    onUpcomingLabelClick: (Int) -> Unit,
+    onNavigateToMovieDetails: (Int) -> Unit,
+    modifier: Modifier
+) {
+    LazyColumn(
+        verticalArrangement = Arrangement.spacedBy(dimensionResource(id = R.dimen.home_screen_spacing))
+    ) {
+        item {
+            PopularMovies(popularViewState, onPopularLabelClick, onNavigateToMovieDetails)
+        }
+
+        item {
+            NowPlayingMovies(nowPlayingViewState, onNowPlayingLabelClick, onNavigateToMovieDetails)
+        }
+
+        item {
+            UpcomingMovies(upcomingViewState, onUpcomingLabelClick, onNavigateToMovieDetails)
+        }
+    }
+}
+
+@Composable
+fun UpcomingMovies(
```

Comment:
The only difference between UpcomingMovies, NowPlayingMovies and PopularMovies is title, so instead of having 3 methods you can just pas title as parameter.